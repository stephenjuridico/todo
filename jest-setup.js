jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

// Fake timers using Jest
beforeEach(() => {
  jest.useFakeTimers();
});
