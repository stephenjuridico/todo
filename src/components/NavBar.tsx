import React from 'react';
import {HStack, IconButton} from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import {useNavigation} from '@react-navigation/native';

const NavBar = () => {
  const navigation = useNavigation();
  return (
    <HStack alignItems="center" alignContent="center">
      <IconButton
        testID="btnGoBack"
        onPress={() => navigation.goBack()}
        borderRadius={'full'}
        _icon={{
          as: Feather,
          name: 'arrow-left',
          size: 6,
          // color: 'white',
        }}
      />
    </HStack>
  );
};

export default NavBar;
