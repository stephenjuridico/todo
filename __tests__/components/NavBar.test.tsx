import 'react-native';
import React from 'react';
import NavBar from '../../src/components/NavBar';
import {fireEvent, render, screen} from '../../src/test/test-utils';
import {mockedGoBack} from '../../jest-setup';

it('renders correctly', () => {
  render(<NavBar />);
  expect(screen.toJSON()).toMatchSnapshot();
});

it('renders correctly and can go back', () => {
  render(<NavBar />);
  const btn = screen.getByTestId('btnGoBack');
  fireEvent.press(btn);
  expect(mockedGoBack).toBeCalled();
});
