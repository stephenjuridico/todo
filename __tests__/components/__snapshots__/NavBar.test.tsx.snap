// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderRadius": 9999,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
      testID="btnGoBack"
    >
      <Text
        allowFontScaling={false}
        dataSet={{}}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "color": "#e6b200",
              "fontSize": 24,
              "height": 24,
              "lineHeight": 24,
              "width": 24,
            },
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ÔÑê
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
